src/twisted/internet/test/test_iocp.py

```diff
@@ -64,16 +69,47 @@ def _acceptAddressTest(self, family, localhost):
         client.setblocking(False)
         try:
             client.connect((localhost, port.getsockname()[1]))
-        except error as e:
+        except OSError as e:
             self.assertIn(e.errno, (errno.EINPROGRESS, errno.EWOULDBLOCK))
 
         server = socket(family, SOCK_STREAM)
         self.addCleanup(server.close)
         buff = array("B", b"\0" * 256)
         self.assertEqual(0, _iocp.accept(port.fileno(), server.fileno(), buff, None))
-        server.setsockopt(
-            SOL_SOCKET, SO_UPDATE_ACCEPT_CONTEXT, pack("P", port.fileno())
-        )
+
+        for attemptsRemaining in reversed(range(5)):
```

Comment:
Since the attempts + delay regulate the number of unintended failures, why not set the values a bit higher? For example a 10 second or even 30 second timeout is fine, since that worst case is only hit when the test fails and even then only when it fails in a very particular way (`accept()` succeeding but the socket never reaching connected state).

Reply:
I think that this test should not take more than 5 seconds to executes.

If it takes 10 seconds for an IOCP socket to accept an already triggered connection, then I think that something is very wrong.

Most of the time this is an instant operation.... so anything about 5 second should raise a warning.

I wanted to keep lower values to see how often it fails with a timeout below 1 second.

Reply:
A test case taking unusually long before passing can indeed be an indication of a deeper problem, but if we are interested in measuring that, it would be better to make that part of the test framework rather than individual tests, in my opinion.

I have no data to support my gut feeling that a 10 seconds timeout is better than 1 second, so let's leave it as it is and see what happens.